File: src/test/java/utils/bookingData.js
function bookingData(){
    var LocalDate = Java.type('java.time.LocalDate');

    function pick(arr){
        return arr[Math.floor(Math.random() * arr.length) ]
    }

    function randInt (min, max){
        return Math.floor(Math.random() * (max - min + 1)) + min;
    }

    function build(overrides){
        var names = ['Adrian', 'Manuel', 'Pedro', 'Luis'];
        var last = ['Matos', 'Oliva', 'Sanchez', 'Soto']

        var checkin = LocalDate.now().toString();
        var checkout = LocalDate.now().plusDays(7).toString();

        var payload = {
            firstname: pick(names),
            lastname: pick(last),
            totalprice: randInt(100, 5000),
            depositpaid: true,
            bookingdates: {
                checkin: checkin,
                checkout: checkout
            },
            additionalneeds: 'Breakfast'
        };

        overrides = overrides || {};

        if (overrides.bookingdates){
            payload.bookingdates.checkin = overrides.bookingdates.checkin || payload.bookingdates.checkin;
            payload.bookingdates.checkout = overrides.bookingdates.checkout || payload.bookingdates.checkout;
            delete overrides.bookingdates;
        }

        for (var k in overrides){
            if (overrides.hasOwnProperty(k)){
                payload[k] = overrides[k];
            }
        }

        return payload;
    }

    return {build: build};
}

File: src/test/java/utils/BookingData.java
package utils;

import java.time.LocalDate;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.ThreadLocalRandom;

public class BookingData {

    private static final String[] NOMBRES = {
            "Adrian", "Luis", "Andres", "Carlos", "Miguel", "Diego", "Jorge"
    };

    private static final String[] APELLIDOS = {
            "Matos", "Pajuelo", "Orlano", "Ibañez", "Gomez", "Ramos", "Torres"
    };

    public static Map<String, Object> build() {

        String nombre = NOMBRES[ThreadLocalRandom.current().nextInt(NOMBRES.length)];
        String apellido = APELLIDOS[ThreadLocalRandom.current().nextInt(APELLIDOS.length)];

        int totalPrice = ThreadLocalRandom.current().nextInt(100, 5000);

        LocalDate checkin = LocalDate.now();
        LocalDate checkout = checkin.plusDays(7);

        Map<String, Object> bookingDates = new HashMap<>();
        bookingDates.put("checkin", checkin.toString());
        bookingDates.put("checkout", checkout.toString());

        Map<String, Object> payload = new HashMap<>();
        payload.put("firstname", nombre);
        payload.put("lastname", apellido);
        payload.put("totalprice", totalPrice);
        payload.put("depositpaid", true);
        payload.put("bookingdates", bookingDates);
        payload.put("additionalneeds", "Breakfast");

        return payload;
    }
}
